src/url.js

```diff
@@ -179,8 +180,15 @@ export function serializeQueryString(params) {
     const v = params[k];
     if (v == null) {
       continue;
+    } else if (isArray(v)) {
+      for (let i = 0; i < v.length; i++) {
+        const sv = /** @type {string} */ (v[i]);
+        s.push(`${encodeURIComponent(k)}=${encodeURIComponent(sv)}`);
```

Comment:
Rails have similar expectation (also for `param[key1][key2]` multi-level object construction.